src/components/user/services/userSession.service.js

```diff
@@ -0,0 +1,49 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.user').provider('UserSessionSrv',
+        function () {
+            'ngInject';
+
+            var isLastSessionRecordDisabled = false;
+            this.disableLastSessionRecord = function (isDisbaled) {
+                isLastSessionRecordDisabled = !!isDisbaled;
+            };
+
+            this.$get = function (InfraConfigSrv, ENV) {
+                'ngInject';// jshint ignore:line
+
+                var initProm,lastSessionData;
+
+                var UserSessionSrv = {};
+
+                UserSessionSrv.isLastSessionRecordDisabled = function () {
+                    return isLastSessionRecordDisabled;
+                };
+
+                UserSessionSrv.getLastSessionData = function () {
+                    return initProm.then(function(){
+                        return lastSessionData;
+                    });
+                };
+
+                function init() {
+                    return InfraConfigSrv.getUserData().then(function (userData) {
+                        var globalLastSessionRef = new Firebase(ENV.fbDataEndPoint + ENV.firebaseAppScopeName + '/lastSessions/' + userData.uid, ENV.firebaseAppScopeName);
+                        return globalLastSessionRef.once('value').then(function(snapshot){
+                            lastSessionData = snapshot.val();
+                            if(!isLastSessionRecordDisabled){
+                                globalLastSessionRef.child('began').set(Firebase.ServerValue.TIMESTAMP);
```

Comment:
firebase set returns a success/failure callback....(or promise)
No need for it?